modules/local/make_pairs_file/main.nf

```diff
@@ -0,0 +1,63 @@
+process MAKE_PAIRS_FILE {
+    tag "${meta.id}"
+    label "process_low"
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/ac/acf7567da73ff97edaab983f28f434826d242cc5684429724f23ad0b0ce99f5f/data' :
+        'community.wave.seqera.io/library/samtools_yahs_gawk:0e36292a5b98b34c' }"
+
+    input:
+    tuple val(meta), path(scaffolds_fai), path(scaffolds_agp), path(contigs_fai), path(contigs_bin)
+
+    output:
+    tuple val(meta), path("*.pairs.gz"), emit: pairs
+    path "versions.yml"                , emit: versions
+
+    script:
+    def prefix        = task.ext.prefix ?: "${meta.id}"
+    """
+    gawk '
+        BEGIN {
+            OFS = "\t"
+            print "## pairs format v1.0"
+        }
+        { print "#chromsize:", \$1, \$2 }
+        END {
+            print "#columns:", "readID", "chr1", "pos1", "chr2", "pos2", "strand1", "strand2"
+        }' ${scaffolds_fai} > ${prefix}.pairs
+
+    juicer pre \\
+        ${contigs_bin} \\
+        ${scaffolds_agp} \\
+        ${contigs_fai} |\\
+        gawk '
+            BEGIN { OFS = "\t" }
+            \$3 > 0 && \$7 > 0 {print ".", \$2, \$3, \$6, \$7, ".", "."}
+        ' |\\
+        LC_ALL=C sort -k2,2d -k4,4d -S50G >>\\
+        ${prefix}.pairs
```

Comment:
so that we better see the list of commands
```suggestion
        ${contigs_fai} |\\
    gawk '
        BEGIN { OFS = "\t" }
        \$3 > 0 && \$7 > 0 {print ".", \$2, \$3, \$6, \$7, ".", "."}
    ' |\\
    LC_ALL=C sort -k2,2d -k4,4d -S50G >>\\
    ${prefix}.pairs
```